src/lib/storage/dictionary_segment.hpp

```diff
@@ -25,48 +32,85 @@ class DictionarySegment : public BaseSegment {
   /**
    * Creates a Dictionary segment from a given value segment.
    */
-  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment);
+  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment) {
+    const auto& values = std::static_pointer_cast<ValueSegment<T>>(base_segment)->values();
+
+    _dictionary = std::make_shared<std::vector<T>>(values.cbegin(), values.cend());
+
+    // Sort the actual values and remove duplicates
```

Comment:
```suggestion
```

Reply:
Hm yea, the sort is pretty straightforward. I wanted to explain the duplicate-removal, since the nifty one-liner might be a bit confusing when skipping over it.

Reply:
I would leave the comment if it's not clear that it removes duplicates.